integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioEnabled.java

```diff
@@ -0,0 +1,402 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static oracle.weblogic.kubernetes.TestConstants.ADMIN_SERVER_NAME_BASE;
+import static oracle.weblogic.kubernetes.TestConstants.MANAGED_SERVER_NAME_BASE;
+import static oracle.weblogic.kubernetes.TestConstants.OCIR_SECRET_NAME;
+import static oracle.weblogic.kubernetes.actions.ActionConstants.MODEL_DIR;
+import static oracle.weblogic.kubernetes.actions.TestActions.addLabelsToNamespace;
+import static oracle.weblogic.kubernetes.actions.TestActions.execCommand;
+import static oracle.weblogic.kubernetes.actions.TestActions.shutdownManagedServerUsingServerStartPolicy;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkServiceExists;
+import static oracle.weblogic.kubernetes.utils.ConfigMapUtils.createConfigMapAndVerify;
+import static oracle.weblogic.kubernetes.utils.DomainUtils.createDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.createMiiImageAndVerify;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.createOcirRepoSecret;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.dockerLoginAndPushImageToRegistry;
+import static oracle.weblogic.kubernetes.utils.IstioUtils.createIstioDomainResource;
+import static oracle.weblogic.kubernetes.utils.OperatorUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.PodUtils.checkPodDoesNotExist;
+import static oracle.weblogic.kubernetes.utils.PodUtils.checkPodReady;
+import static oracle.weblogic.kubernetes.utils.SecretUtils.createSecretWithUsernamePassword;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@DisplayName("Test WLS Session Migration via istio enabled")
+@IntegrationTest
+class ItIstioSessionMigrIstioEnabled {
+
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+
+  // constants for creating domain image using model in image
+  private static final String SESSMIGR_MODEL_FILE = "model.sessmigr.yaml";
+  private static final String SESSMIGR_IMAGE_NAME = "sessmigr-mii-image";
+
+  // constants for web service
+  private static final String SESSMIGR_APP_NAME = "sessmigr-app";
+  private static final String SESSMIGR_APP_WAR_NAME = "sessmigr-war";
+  private static final int SESSION_STATE = 4;
+  private static Map<String, String> httpAttrMap;
+
+  // constants for operator and WebLogic domain
+  private static String domainUid = "sessmigr-domain-1";
```

Comment:
Pani, Since all three session migration test suites share same yaml file, I can't change the domainUID, since all three test suites share the same yaml file, they have to use same domain name

Reply:
Can we use a template model file  to generate the real model file to be used. I want to avoid any intermittent failure in parallel run when same domain id is being used in multiple class

Reply:
modified the yaml file on the fly. Each test now gets its own yaml file and we can have diff domain UID for each test, that's our purpose